lib/extensions/ar_migration.rb

```diff
@@ -1,6 +1,6 @@
 module ArPglogicalMigrationHelper
-  def self.discover_schema_migrations_ran_class
-    return unless ActiveRecord::Base.connection.table_exists?("schema_migrations_ran")
+  def self.discover_schema_migrations_ran_class(version)
+    return unless version > "20171031010000"
```

Comment:
I think this has to be `>=` so that it runs at the tail end of https://github.com/ManageIQ/manageiq-schema/blob/master/db/migrate/20171031010000_add_schema_migrations_ran.rb

Reply:
(we should check what the original did - whether it ran as part of this migration or the one right afterwards)

Reply:
yeah, it blows up on the down when I change this to `>=`.  I think it's because it's called from both before and after the migration, so the only time that both the up and down direction have the `schema_migrations_ran` table is at versions greater than this...

Either way, I think doing this is safe and prevents excessive queries for versions less than this version...

```ruby 
return if version <= "20171031010000" && !ActiveRecord::Base.connection.table_exists?("schema_migrations_ran")
```

Reply:
> I think it's because it's called from both before and after the migration,

See below, it's called before and after migrations:
```ruby
module ArPglogicalMigration
  def migrate(direction)
    PglogicalSubscription.all.each do |s|
      ### BEFORE
      ArPglogicalMigrationHelper::RemoteRegionMigrationWatcher.new(s, version.to_s).wait_for_remote_region_migration
    end
    ret = super
    ### AFTER
    ArPglogicalMigrationHelper.update_local_migrations_ran(version.to_s, direction)
    ret
  end
end
```